pkg/resource/deploy/plan_executor.go

```diff
@@ -148,19 +145,35 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 					// Signal completion to the step executor. It'll exit once it's done retiring all of the steps in
 					// the chain that we just gave it.
 					pe.stepExec.SignalCompletion()
-					log.Infof("planExecutor.Execute(...): issued deletes")
+					logging.V(4).Infof("planExecutor.Execute(...): issued deletes")
 
 					return false, nil
 				}
 
-				if eventErr := pe.handleSingleEvent(event.Event); eventErr != nil {
-					log.Infof("planExecutor.Execute(...): error handling event: %v", eventErr)
-					pe.reportError(pe.plan.generateEventURN(event.Event), eventErr)
+				flow, eventErr := pe.handleSingleEvent(event.Event)
+				if eventErr != nil || flow == Bail {
+					if eventErr != nil {
+						logging.V(4).Infof("planExecutor.Execute(...): error handling event: %v", eventErr)
+						pe.reportError(pe.plan.generateEventURN(event.Event), eventErr)
+					}
+
+					logging.V(4).Infof("planExecutor.Execute(...): canceling context")
 					cancel()
```

Comment:
this was an issue before.  However, i think it would probably always be good to explain the implications of these sorts of calls as htey are essentially non local flow control as well.

Reply:
will do.